lia-ui/src/main/java/com/lithium/community/android/ui/components/adapters/LiConversationAdapter.java

```diff
@@ -169,136 +185,317 @@ private void showPopupMenu(final View view, final int position) {
+                builder.setMessage(R.string.li_are_you_sure)
+                        .setPositiveButton(R.string.li_yes, dialogClickListener)
+                        .setNegativeButton(R.string.li_no, dialogClickListener).show();
+
+                return true;
+            } else if (menuItem.getItemId() == R.id.li_action_subscribe) {
+                boolean isSubscribed = item.getUserContext().isSubscribed();
+                if (isSubscribed) {
+                    unsubscribe(item, position);
+                } else {
+                    subscribe(item, position);
+                }
+                return false;
+            } else {
+                return false;
+            }
+        }
+    }
 
-                        LiClientRequestParams.LiMarkMessageParams params = new LiClientRequestParams.LiMarkMessageParams(activity,
-                                String.valueOf(item.getAuthor().getId()),
-                                String.valueOf(item.getId()), isMarkUnread);
-                        try {
-                            LiClientManager.getMarkMessagePostClient(params).processAsync(new LiAsyncRequestCallback<LiPostClientResponse>() {
-                                @Override
-                                public void onSuccess(LiBaseRestRequest request, LiPostClientResponse response) throws LiRestResponseException {
-                                    if (response.getHttpCode() == LiCoreSDKConstants.HTTP_CODE_SUCCESSFUL) {
-                                        item.getUserContext().setRead(!item.getUserContext().getRead());
-                                        LiUIUtils.showMarkReadSuccessful(activity, item);
-                                    } else {
-                                        LiUIUtils.showMarkReadUnSuccessful(activity, item);
-                                    }
-                                }
+    private void openReportAbuse(LiMessage item) {
+        Bundle bundle = new Bundle();
+        bundle.putString(LiSDKConstants.REPORT_ABUSE_MESSAGE_ID, item.getId() + "");
+        bundle.putString(LiSDKConstants.REPORT_ABUSE_AUTHOR_ID, item.getAuthor().getId() + "");
+        Intent i = new Intent(activity, LiReportAbuseActivity.class);
+        i.putExtras(bundle);
+        activity.startActivity(i);
+    }
 
-                                @Override
-                                public void onError(Exception exception) {
-                                    LiUIUtils.showMarkReadUnSuccessful(activity, item);
-                                }
-                            });
-                        } catch (LiRestResponseException e) {
-                            Log.d(LiSDKConstants.GENERIC_LOG_TAG, e.getMessage());
+    private void markRead(LiMessage item) {
+        if (item.getUserContext() != null) {
+            boolean isMarkUnread = false;
+            if (item.getUserContext() != null && item.getUserContext().getRead()) {
```

Comment:
This check is redundant as we have already established that item.getUserContext() is null. The code can be condensed to
isMarkUnread = item.getUserContext().getRead();

Reply:
sure